src/Raven.Client/ServerWide/DatabaseTopology.cs

```diff
@@ -412,6 +419,9 @@ public void RemoveFromTopology(string delDbFromNode)
             {
                 if (Members.Contains(mentorNode))
                     return mentorNode;
+                
+                if (AllNodes.Contains(mentorNode) && task.IsPinnedToMentorNode())
```

Comment:
Here the `MentorNode` has precedence over the pinned one.. is that what we want?

Reply:
Also we enter here only if `mentorNode != null`, is that okay?

Reply:
The `MentorNode` _is_ the pinned node.

Reply:
Right. so all looks good here.